recipes-security/optee/optee-os/0001-imx7ulp-watchdog-add-initialization-function.patch

```diff
@@ -0,0 +1,64 @@
+From 3614eb7282e35b3e2ac02aa5b58424b03ab0d5a0 Mon Sep 17 00:00:00 2001
+From: Jorge Ramirez-Ortiz <jorge@foundries.io>
+Date: Tue, 10 Dec 2019 21:21:02 +0100
+Subject: [PATCH] imx7ulp: watchdog: add initialization function.
+
+atarting the watchdog clock allows it to operate without any normal
+world initialization.
+
+This could be the case on other IMX platforms.
+
+Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
+---
+ core/drivers/imx_wdog.c | 24 +++++++++++++++++++++++-
+ 1 file changed, 23 insertions(+), 1 deletion(-)
+
+diff --git a/core/drivers/imx_wdog.c b/core/drivers/imx_wdog.c
+index 83f461c2..293e929f 100644
+--- a/core/drivers/imx_wdog.c
++++ b/core/drivers/imx_wdog.c
+@@ -44,6 +44,20 @@
+ static bool ext_reset;
+ static vaddr_t wdog_base;
+ 
++#ifdef CFG_MX7ULP
++static void imx_wdog_initialize(void)
++{
++	/* unlock the wdog for reconfiguration */
```

Comment:
Can you add an extra check for wdog_base here? Just to be safe in case code ends up moving around later on.